libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/PredictiveCache.kt

```diff
@@ -11,14 +11,14 @@ object PredictiveCache {
     internal val cachedNavigationPredictiveCacheControllers =
         mutableSetOf<PredictiveCacheController>()
     internal val cachedMapsPredictiveCacheControllers =
-        mutableMapOf<Any, Pair<TilesetDescriptor, PredictiveCacheController>>()
+        mutableMapOf<Any, List<Pair<TilesetDescriptor, PredictiveCacheController>>>()
     internal val cachedMapsPredictiveCacheControllersTileVariant =
         mutableMapOf<Any, MutableMap<String, PredictiveCacheController>>()
 
     internal val navPredictiveCacheLocationOptions =
         mutableSetOf<PredictiveCacheLocationOptions>()
-    internal val mapsPredictiveCacheLocationOptions =
-        mutableMapOf<Any, Triple<TilesetDescriptor, TileStore, PredictiveCacheLocationOptions>>()
+    internal val mapsPredictiveCacheLocationOptions = mutableMapOf<Any,
+        Pair<TileStore, List<Pair<TilesetDescriptor, PredictiveCacheLocationOptions>>>>()
     internal val mapsPredictiveCacheLocationOptionsTileVariant =
         mutableMapOf<Any, MutableMap<String, Pair<TileStore, PredictiveCacheLocationOptions>>>()
```

Comment:
I'm thinking if we need to store `TileStore` here at all.
We use it to recreate controllers, but `TileStore` should be the same instance across all calls `createMapsController`. We can save it as a PredictiveCache::tileStore on initial call and use for recreations

Reply:
Should it? As far as I understand, for different `MapboxMap`s there can be different `TIleStore`s. That's how we obtain TileStore:
```
val tileStore = map.getResourceOptions().tileStore
```

Reply:
tileStore instance should be built on a customer side and the instance should be passed to `Maps, NN, Search`, so all SDKs will work with the same offline folder
`val tileStore = TileStore.create(context.provideTileStorePath())`
according to the docs 
```
Creates a TileStore instance for the given storage path. The returned instance exists as long as it is retained by the client. If the tile store instance already exists for the given path this method will return it without creating a new instance, thus making sure that there is only one tile store instance for a path at a time. If the given path is empty, the tile store at the default location is returned. 
```
that's why I think we shouldn't store it for each MapboxMap

Reply:
what for?

Reply:
I don't know, all I'm saying is that it's possible, so should we artificially remove this possibility? Especially considering the fact that it was possible previously.

Reply:
Anyway, I think we can follow-up separately as this PR did not change this behaviour, right?